notebooks/11 Pandas.ipynb

```diff
@@ -613,11 +1197,248 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 33,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "E\n",
+       "Class1    12\n",
+       "Class2     7\n",
+       "Name: A, dtype: int64"
+      ]
+     },
+     "execution_count": 33,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "df.groupby('E')['A'].min()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Working with dates\n",
+    "\n",
+    "In the sample data set you have worked with in this notebook, the column named 'F' contains strings which represent dates. As you can see, these strings use two digits for the days, two digits for the months and four digits for the years. These components are separated by hyphens. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 34,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "    A   B   C     D       E           F     G\n",
+      "0  12   5   6  12.0  Class1  01-05-2022  15.0\n",
+      "1  14  11  17  20.0  Class1  01-06-2022  23.0\n",
+      "2  15   6   8  19.0  Class1  01-07-2022  22.0\n",
+      "3   8   3  21   5.0  Class2  01-08-2022   8.0\n",
+      "4  10   9  14   7.0  Class2  01-09-2022  10.0\n",
+      "5   7   5  16   NaN  Class2  01-10-2022   NaN\n"
+     ]
+    }
+   ],
+   "source": [
+    "print( df )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "When you want to work with dates in Pandas, it is generally best to save these not as basic strings, but, rather, as `datetime` objects. When the dates are saved in this specific  format, it becomes much easier to analyse or to compare dates. \n",
```

Comment:
You generally don't *save* dataframes as Pandas objects, so maybe rephrase this sentence like this:
```suggestion
    "When you want to work with dates in Pandas, it is generally best to interpret these not as basic strings, but, rather, as `datetime` objects. When the dates are interpreted in this specific data type, it becomes much easier to analyse or to compare dates.\n",
```